### DISCUSSION TITLE
How to access the local time and/or use UTC time?

### DISCUSSION THREAD

**Discussion by Tungstic** [wojtekmaj/react-date-picker#624] [Q&A] [answered]
I'm building a Next.js full-stack app and want to save the date from Date-Picker in my database. At first, I tried to work only with dates but there was always a 1-day difference between my date in the localhost and the date sent to the database. I realized that new Date() is a timestamp so I might as well use timestamptz for my data format in the database.

Now there's a 4-hour difference between my time in the localhost and the time sent to my database.
I've read dozens of explanations of the date issues in JS which all boil down to having the correct time zone (the country with DST also has a different time zone offset depending on whether it's summer or winter). But I can't figure out how to fix the time difference. Can somebody please help me?

PS: hasMounted state is used so that there's no hydration error on reload

```export default function EntryForm(props) {
 
  const [mealDate, setMealDate] = useState(Date());
  const [hasMounted, setHasMounted] = useState(false);

  useEffect(() => setHasMounted(true), []);
return
<other elements>
  {hasMounted === true && (
        <DatePicker
          aria-label="pick the entry's date"
          locale="en-US"
          onChange={setMealDate}
          value={mealDate}
        />
      )}

**Comment by wojtekmaj** [comment, marked as answer]
React-Date-Picker returns local, native JavaScript Date. It's likely not "one day off", it's rather "In Greenwich, it's still yesterday/already tomorrow at this time".

My personal advice would be to keep it as it is, because working with anything else than UTC will be the _start_, not the _end_ of your timezones nightmare.
